Dockerfile

```diff
@@ -0,0 +1,13 @@
+FROM golang:1.17.6 as build
+
+WORKDIR /go/src/app
+ENV CGO_ENABLED=0
+ADD . /go/src/app
+
+RUN go get -d -v ./...
+RUN go build -ldflags '-s -w' -o /go/bin/ntopng-exporter
```

Comment:
Usually, when building a go app into a minimal container I also add `-extldflags "static"` to the `-ldflags` option. Maybe since this is a distroless container instead of a scratch container this isn't necessary? Or maybe its because it already isn't linking to any shared libraries due to its function? It seems like it works both with and without it, but if you don't know a reason not to include it, I would default to including it.

Reply:
I looked at that, and apparently the `-extldflags "static"` option has no effect when cgo is enabled. It will always output dynamically linked libraries. The environment variable `CGO_ENABLED=0` does the trick though. This is not great usability :(

Reply:
Very interesting. I have to admit you prompted me to do some research here myself and I found the same thing. I guess I never looked into cgo enable / disable enough to understand that all places where cgo is disabled are static by default. But when thinking about it critically it makes sense.

Thanks for pointing me to another new thing today 😉